pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -146,118 +146,245 @@ func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStart
-			return fmt.Errorf("failed getting mandatory FlowSchema %s due to %v, will retry later", expectedFlowSchema.Name, err)
-		}
-		_, err = flowcontrolClientSet.FlowSchemas().Create(context.TODO(), expectedFlowSchema, metav1.CreateOptions{FieldManager: thisFieldManager})
-		if apierrors.IsAlreadyExists(err) {
-			klog.V(3).Infof("Mandatory FlowSchema %s already exists, skipping creating", expectedFlowSchema.Name)
-			continue
-		}
-		if err != nil {
-			return fmt.Errorf("cannot create mandatory FlowSchema %s due to %v", expectedFlowSchema.Name, err)
-		}
-		klog.V(3).Infof("Created mandatory FlowSchema %s", expectedFlowSchema.Name)
-	}
-	for _, expectedPriorityLevelConfiguration := range priorityLevels {
-		actualPriorityLevelConfiguration, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(context.TODO(), expectedPriorityLevelConfiguration.Name, metav1.GetOptions{})
-		if err == nil {
-			// TODO(yue9944882): extract existing version from label and compare
-			// TODO(yue9944882): create w/ version string attached
-			wrongSpec, err := priorityLevelHasWrongSpec(expectedPriorityLevelConfiguration, actualPriorityLevelConfiguration)
-			if err != nil {
-				return fmt.Errorf("failed checking if mandatory PriorityLevelConfiguration %s is up-to-date due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-			}
-			if wrongSpec {
-				if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Update(context.TODO(), expectedPriorityLevelConfiguration, metav1.UpdateOptions{FieldManager: thisFieldManager}); err != nil {
-					return fmt.Errorf("failed upgrading mandatory PriorityLevelConfiguration %s due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-				}
-				klog.V(3).Infof("Updated mandatory PriorityLevelConfiguration %s because its spec was %#+v but must be %#+v", expectedPriorityLevelConfiguration.Name, actualPriorityLevelConfiguration.Spec, expectedPriorityLevelConfiguration.Spec)
-			}
-			continue
-		}
-		if !apierrors.IsNotFound(err) {
-			return fmt.Errorf("failed getting PriorityLevelConfiguration %s due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-		}
-		_, err = flowcontrolClientSet.PriorityLevelConfigurations().Create(context.TODO(), expectedPriorityLevelConfiguration, metav1.CreateOptions{FieldManager: thisFieldManager})
-		if apierrors.IsAlreadyExists(err) {
-			klog.V(3).Infof("Mandatory PriorityLevelConfiguration %s already exists, skipping creating", expectedPriorityLevelConfiguration.Name)
-			continue
+// fsSuggestedUpdateStrategy determines the auto-update strategy for the mandatory FlowSchema objects
+func fsMandatoryUpdateStrategy(current, desired *flowcontrolv1beta1.FlowSchema) (*flowcontrolv1beta1.FlowSchema, bool) {
+	if current == nil || desired == nil {
+		return nil, false
+	}
+
+	if specChanged := flowSchemaSpecChanged(desired, current); !specChanged {
+		return nil, false
+	}
+
+	copy := desired.DeepCopy()
```

Comment:
Why copy more than the Spec?
Why even copy the Spec?

Reply:
I guess we don't need `DeepCopy` for the bootstrap object, I am deep copying the spec now, let me know if this okay, otherwise I can remove the deep copy.